CargoMonitor/CargoInfoReader.cs

```diff
@@ -0,0 +1,146 @@
+﻿using EddiDataDefinitions;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading;
+using Utilities;
+
+
+
+namespace EddiCargoMonitor
+{
+    public class CargoInfoReader
+    {
+        public int Count { get; set; }
+        public List<CargoInfo> Inventory { get; set; }
+
+        public CargoInfoReader()
+        {
+            Inventory = new List<CargoInfo>();
+        }
+
+        public static CargoInfoReader FromFile(string filename = null)
+        {
+            CargoInfoReader info = new CargoInfoReader();
+            Regex Filter = new Regex(@"^Cargo\.json$");
```

Comment:
It's a fair question I think. Proposed fix... add a new method to `Files.cs` in the Utilities project (since both StatusMonitor and CargoMonitor share such similar code) :
```cs
        // Obtain file info for a file name and path, or null if the file is not available
        public static FileInfo FileInfo(string path, string file)
        {
            if (path == null)
            {
                // Configuration can be changed underneath us so we do have to check each time...
                return null;
            }

            try
            {
                FileInfo info = new FileInfo(path + @"\" + file);
                if (info.Exists)
                {
                    // This info can be cached so force a refresh
                    info.Refresh();
                }
                return info;
            }
            catch { return null; }
        }
```
This removes the directory search and the regex, and nothing appears to be broken, so I'm going to go ahead and incorporate it into my branch. :-)